src/utils.js

```diff
@@ -2,16 +2,25 @@
  * @ignore
  */
 export default class {
-    getPerformanceInfo() {
-        let timing = window.performance.timing,
-            interactive = timing.domInteractive - timing.domLoading,
-            dcl = timing.domContentLoadedEventStart - timing.domLoading,
-            complete = timing.domComplete - timing.domLoading;
-        return {
-            ptInteractive: interactive >= 0 ? interactive : undefined,
-            ptDcl: dcl >= 0 ? dcl : undefined,
-            ptComplete: complete >= 0 ? complete : undefined
-        };
+    getPerformanceInfo(targetWindow) {
+        if('navigation' in window[targetWindow].performance && 'timing' in window[targetWindow].performance){
```

Comment:
I prefer to extract `window[targetWindow].performance` as a var for readability and a slightly bit of performance

Reply:
Good point. However, it sounds like the scope is not just for this PR. I think we should create an issue or another PR for the refactoring(optimization) for the implementation of `window[targetWindow]`. An objective of this PR is to add a support for Navigation Type.